src/Chapter3.hs

```diff
@@ -970,7 +1087,28 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
-
+data DayOfWeek =
+    Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Enum, Bounded, Eq, Ord, Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay d | d == maxBound = minBound
+          | otherwise = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty d = (fromEnum Friday) - (fromEnum d)
```

Comment:
This needs a slight modification. If it is the weekend then you would get the negative numbers in here.

Reply:
Nice catch, I'll fix that up